tests/system/model_monitoring/test_app.py

```diff
@@ -431,7 +431,8 @@ def custom_setup_class(cls) -> None:
         cls.run_db = mlrun.get_run_db()
 
     def custom_setup(self) -> None:
-        _V3IORecordsChecker.custom_setup(project_name=self.project_name)
+        # skips TestMLRunSystem, calls custom_setup() of _V3IORecordsChecker
+        super(TestMLRunSystem, self).custom_setup(project_name=self.project_name)
```

Comment:
Or we can refactor this to _not_ use multiple inheritance.

Reply:
I'm not sure either, but I guess there are pros and cons to this. Can be discussed in one of our meetings f2f.

Reply:
I'm pretty sure the comment isn't correct btw. Why do you think it skips `TestMLRunSystem`? `TestMLRunSystem` is the one that defines the logger.

Reply:
Now entirely sure - but still few references:

1. See python [docs](https://docs.python.org/3/library/functions.html#super) for `super`:
>The object_or_type determines the [method resolution order](https://docs.python.org/3/glossary.html#term-method-resolution-order) to be searched. The search starts from the class right after the type.

2. From another view, looking at `TestMLRunSystem` custom_setup() you can see it has a different signature (without project_name) so it would have raised an exception if called from here.

3. As I understand it, the reason the change is working is that it calls custom_setup() on the TestMonitoringAppFlow class, which has it's _logger class attribute set from the TestMLRunSystem inherited class. On the hand, the original code called _V3IORecordsCheck custom_setup() - and on that class the _logger attribute in not initialized.